src/Activity.js

```diff
@@ -0,0 +1,74 @@
+class Activity {
+  constructor(activityData) {
+    this.activityData = activityData;
+  }
+
+  getUserData(id) {
+    return this.activityData.filter(obj => obj.userID === id);
+  }
+
+  getDay(id, date) {
+    return this.getUserData(id).find(obj => obj.date === date);
+  }
+
+  getWeek(id, date) {
+    let endDate = this.getUserData(id).findIndex(obj => obj.date === date);
+    return endDate - 6 >= 0 ?
+      this.getUserData(id).slice(endDate - 6, endDate + 1) :
+      this.getUserData(id).slice(0, endDate + 1);
+  }
+
+  getMilesByDay(id, date, userRepo) {
+    let feetWalked = this.getDay(id, date).numSteps * userRepo.getUserData(id).strideLength;
+    return +(feetWalked / 5280).toFixed(1);
+  }
+
+  getMinutesByDay(id, date) {
+    return this.getDay(id, date).minutesActive;
+  }
+
+  getAverageMinutesByWeek(id, date) {
+    let weekData = this.getWeek(id, date);
+    return Math.round(weekData.reduce((acc, day) =>
+      acc + day.minutesActive, 0) / weekData.length);
+  }
+
+  checkStepGoal(id, date, userRepo) {
+    return this.getDay(id, date).numSteps > userRepo.getUserData(id).dailyStepGoal;
+  }
+
+  getGoalDays(id, userRepo) {
+    return this.getUserData(id).filter(obj =>
+      obj.numSteps > userRepo.getUserData(id).dailyStepGoal);
+  }
+
+  getStairRecord(id) {
+    return this.getUserData(id).sort((a, b) =>
+      a.flightsOfStairs - b.flightsOfStairs).pop().flightsOfStairs;
+  }
+
+  getAverageStairsByDay(date) {
+    let userIDs = [...new Set(this.activityData.map(object => object.userID))];
+    let totalStairs = userIDs.reduce((acc, ID) =>
+      acc + this.getDay(ID, date).flightsOfStairs, 0);
+    return Math.round(totalStairs / userIDs.length);
+  }
+
+  getAverageStepsByDay(date) {
+    let userIDs = [...new Set(this.activityData.map(object => object.userID))];
+    let totalSteps = userIDs.reduce((acc, ID) =>
+      acc + this.getDay(ID, date).numSteps, 0);
+    return Math.round(totalSteps / userIDs.length);
+  }
+
+  getAverageMinutesByDay(date) {
+    let userIDs = [...new Set(this.activityData.map(object => object.userID))];
+    let totalMinutes = userIDs.reduce((acc, ID) =>
+      acc + this.getDay(ID, date).minutesActive, 0);
+    return Math.round(totalMinutes / userIDs.length);
+  }
```

Comment:
For our metric we could make it easy and just do total feet walked,
we could do miles walked in a week per day(so an array of numbers for a week [1.2, 4.3, etc],
we could also turn something that involves minutes into hours.

Reply:
I like the miles per day for a week, I'll see what I can whip up